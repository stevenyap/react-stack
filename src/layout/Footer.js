/**
 * @flow
 * Responsibility: Render the footer
 */
import React from 'react'
import { Grid, Row, Col } from 'react-bootstrap/lib'

import Colors from 'theme/Colors'

export const Footer = () => {
  return (
    <div style={styles.footer}>
      <Grid>
        <Row>
          <Col>
            Â© React Stack. All rights reserved.
          </Col>
        </Row>
      </Grid>
    </div>
  )
}

const styles = {
  footer: {
    marginTop: 30,
    padding: 10,
    background: Colors.orange,
    color: Colors.white,
    textAlign: 'center'
  }
}

export default Footer
