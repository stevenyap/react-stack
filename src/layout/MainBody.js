/**
 * @flow
 * Responsibility: Render the main body of the app
 */
import React from 'react'
import { Switch, Route } from 'react-router-dom'
import Grid from 'react-bootstrap/lib/Grid'

import Home from 'pages/Home'

export const MainBody = () => {
  return (
    <Grid style={styles.container}>
      <Switch>
        <Route exact={true} path="/" component={Home} />
      </Switch>
    </Grid>
  )
}

const styles = {
  container: {
    flex: '1'
  }
}

export default MainBody
