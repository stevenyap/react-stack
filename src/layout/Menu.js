/**
 * @flow
 * Responsibility: Display the menu
 */
import type { ContextRouter } from 'react-router-dom'

import React from 'react'
import { connect } from 'react-redux'
import { Nav, Navbar, NavItem } from 'react-bootstrap/lib'
import { Link, withRouter } from 'react-router-dom'

import { doLogout } from 'ducks/admin'

type MenuLinkProps = { to: string, text: string } & ContextRouter
const MenuLink = withRouter((props: MenuLinkProps) => {
  const { to, text, history } = props
  const handleClick = () => history.push(to)
  return <NavItem href="#" onClick={handleClick}>{text}</NavItem>
})

type ComponentState = { admin: Admin }
type Props = ComponentState & { ispatch: Dispatch }
export const Menu = (props: Props) => {
  const { admin, dispatch } = props
  const logout = () => dispatch(doLogout())

  return (
    <Navbar inverse={true} collapseOnSelect={true}>
      <Navbar.Header>
        <Navbar.Brand>
          <Link to="/">React Stack</Link>
        </Navbar.Brand>
        {admin && <Navbar.Toggle />}
      </Navbar.Header>
      {admin &&
        <Navbar.Collapse>
          <Nav pullRight>
            <MenuLink to="/" text="Home" />
            <NavItem href="#" onClick={logout}>Logout</NavItem>
          </Nav>
        </Navbar.Collapse>}
    </Navbar>
  )
}

const mapStateToProps = (state: State): ComponentState => ({
  admin: state.admin.data
})
export default connect(mapStateToProps)(Menu)
