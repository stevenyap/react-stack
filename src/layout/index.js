/**
 * @flow
 * Responsibility: Render the layout of the app
 */
import React from 'react'

import Menu from 'layout/Menu'
import MainBody from 'layout/MainBody'
import Footer from 'layout/Footer'

export const Layout = () => {
  return (
    <div style={styles.container}>
      <Menu />
      <MainBody />
      <Footer />
    </div>
  )
}

const styles = {
  container: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh'
  }
}

export default Layout
