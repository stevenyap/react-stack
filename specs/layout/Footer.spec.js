import React from 'react'
import { BrowserRouter } from 'react-router-dom'

import { Footer } from 'layout/Footer'

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderJSON(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    )
    expect(tree).toMatchSnapshot()

    const renderedText = toText(tree)
    expect(renderedText).toContain('All rights reserved')
  })
})
