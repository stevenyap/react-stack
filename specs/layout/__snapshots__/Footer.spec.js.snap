// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  style={
    Object {
      "background": "#f0ad4e",
      "color": "#fff",
      "marginTop": 30,
      "padding": 10,
      "textAlign": "center",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className=""
      >
        Â© React Stack. All rights reserved.
      </div>
    </div>
  </div>
</div>
`;
